code/go/0chain.net/smartcontract/dbs/event/allocation_blobber_term.go

```diff
@@ -0,0 +1,134 @@
+package event
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"gorm.io/gorm"
+)
+
+type AllocationBlobberTerm struct {
+	gorm.Model
+	AllocationID            string        `json:"allocation_id" gorm:"primaryKey; not null"` // Foreign Key
+	BlobberID               string        `json:"blobber_id" gorm:"primaryKey; not null"`    // Foreign Key
+	ReadPrice               int64         `json:"read_price"`
+	WritePrice              int64         `json:"write_price"`
+	MinLockDemand           float64       `json:"min_lock_demand"`
+	MaxOfferDuration        time.Duration `json:"max_offer_duration"`
+	ChallengeCompletionTime time.Duration `json:"challenge_completion_time"`
+	Allocation              Allocation    `json:"-" gorm:"references:AllocationID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
+	Blobber                 Blobber       `json:"-" gorm:"references:BlobberID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
+}
+
+func (edb *EventDb) GetAllocationBlobberTerm(allocationID string, blobberID string) (*AllocationBlobberTerm, error) {
+	if allocationID == "" && blobberID == "" {
+		return nil, fmt.Errorf("can not retrieve term with empty Allocation ID and blobber ID")
+	}
+	var term AllocationBlobberTerm
+	if err := edb.Store.Get().Model(&AllocationBlobberTerm{}).
+		Where(&AllocationBlobberTerm{AllocationID: allocationID, BlobberID: blobberID}).
+		Scan(&term).Error; err != nil {
+		return nil, fmt.Errorf("failed to retrieve term: AllocationID %v and blobberID %v does not exist, error %v",
+			allocationID, blobberID, err)
+	}
+
+	return &term, nil
+}
+
+func (edb *EventDb) GetAllocationBlobberTerms(allocationID, blobberID string) ([]AllocationBlobberTerm, error) {
+	var terms []AllocationBlobberTerm
+	return terms, edb.Store.Get().Model(&AllocationBlobberTerm{}).
+		Where(AllocationBlobberTerm{AllocationID: allocationID, BlobberID: blobberID}).
+		Find(&terms).Error
+}
+
+func (edb *EventDb) deleteAllocationBlobberTerm(allocationID string, blobberID string) error {
```

Comment:
probably deleteAllocationBlobberTermS as we don't validate input parameters

Reply:
Yes, that's more appropriate.

Reply:
@iamrz1 so if we pass empty string it will match for `NULL` is it correct?

Reply:
If by `NULL` you mean `no match`, then that'd be correct.